docs/pages/build-reference/variants.md

```diff
@@ -0,0 +1,99 @@
+
+**If you have a bare project**, you can accomplish this using flavors (Android) and targets (iOS). You can then configure which flavor is used with the `gradleCommand` field on your build profile, and you can configure the target that is used with the `scheme` field for iOS. 
+
+**If you have a managed project**, this can be accomplished by using **app.config.js** and environment variables in **eas.json**.
+
+## Example: configuring development and production variants
+
+Let's say we wanted a development build and production build of our managed Expo project. Your **eas.json** might look like this:
+
+```json
+{
+  "build": {
+    "development": {
+      "developmentClient": true
+    },
+    "production": {}
+  }
+}
+```
+
+And your **app.json** might look like this:
+
+```json
+{
+  "expo": {
+    "name": "MyApp",
+    "slug": "my-app",
+    "ios": {
+      "bundleIdentifier": "com.myapp"
+    },
+    "android": {
+      "package": "com.myapp"
+    }
+  }
+}
+```
+
+Let's convert this to **app.config.js** so we can make it more dynamic:
+
+```json
+export default {
+  name: "MyApp",
+  slug: "my-app",
+  ios: {
+    bundleIdentifier: "com.myapp",
+  },
+  android: {
+    package: "com.myapp",
+  },
+};
+```
+
+Now let's switch out the iOS `bundleIdentifier` and Android `package` (which becomes the Application ID) based on the presence of an environment variable in **app.config.js**:
+
+```js
+const IS_PROD = process.env.APP_VARIANT === "production";
+
+export default {
+  // You can also switch out the app icon to more clearly differentiate the app
+  name: IS_PROD ? "MyApp" : "MyApp (Dev)",
+  slug: "my-app",
+  ios: {
+    bundleIdentifier: IS_PROD ? "com.myapp" : "com.myapp.dev",
+  },
+  android: {
+    package: IS_PROD ? "com.myapp" : "com.myapp.dev",
+  },
+};
+```
+
+> Note: if you are using any libraries that require you to register your application identifier with an external service to use the SDK, such as Google Maps, you will need to have a separate configuration for that API for the iOS Bundle Identifier and Android Package. You can also swap this configuration in using the same approach as above.
+
+To automatically set the `APP_VARIANT` environment variable, we can use `env` in **eas.json**:
+
+```json
+{
+  "build": {
+    "development": {
+      "developmentClient": true
+    },
+    "production": {
+      "env": {
+        "APP_VARIANT": "production"
+      }
+    }
+  }
+}
+```
+
+Now when you run `eas build --profile production` the `APP_VARIANT` environment variable will be set to `"production"` when evaluating **app.config.js** locally and on the EAS Build worker. [Learn more about environment variables on EAS Build](./variables.md).
+
+When you run `expo start` the `APP_VARIANT` variable environment will not be set, and the app will run as the development variant.
```

Comment:
In case of dev client `APP_VARIANT` set durring build will be used to set applicationIde/bundleIdentifier , but values that js can read from expo-constant will be set based `APP_VARIANT` set when running `expo start`

also bundleIdentifier/applicationId values will affect behavior when you click `a` or `i` in expo start to run on the device. (command to lunch the app contains that identifier)